pandas/core/generic.py

```diff
@@ -5909,6 +5909,100 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    def as_nullable_dtypes(
+        self: FrameOrSeries, convert_integer: bool_t = True
+    ) -> FrameOrSeries:
+        """
+        Convert columns of DataFrame or a Series to dtypes supporting ``pd.NA``.
+
+        Parameters
+        ----------
+        convert_integer : bool, default True
+            Whether ``int`` types should be converted to integer extension types.
+
+        Returns
+        -------
+        Series or DataFrame
+            Copy of input object with new dtype.
+
+        Notes
+        -----
+        If the dtype is ``object``, if possible, convert to ``StringDtype``,
```

Comment:
need a convert_string option

Reply:
> need a convert_string option

I am not sure this is needed (although it is easy to add). In contrast to integers (and booleans), we don't already have a numpy string dtype. In case of the integers, there can be advantages of keeping it (more performant, more supported numerical operations). For strings in objects vs string dtype, I think there is less reason to keep it as object .